probe-rs/src/debug/mod.rs

```diff
@@ -442,15 +495,151 @@ impl DebugInfo {
     ) -> StackFrameIterator<'a, 'b> {
         StackFrameIterator::new(&self, core, address)
     }
+
+    /// Find the program counter where a breakpoint should be set,
+    /// given a source file, a line and optionally a column.
+    pub fn get_breakpoint_location(
+        &self,
+        path: &Path,
+        line: u64,
+        column: Option<u64>,
+    ) -> Result<Option<u64>, Error> {
+        debug!(
+            "Looking for breakpoint location for {}:{}{}",
+            path.display(),
+            line,
+            column
+                .map(|c| c.to_string())
+                .unwrap_or_else(|| "-".to_owned())
+        );
+
+        let mut unit_iter = self.dwarf.units();
+
+        let mut locations = Vec::new();
+
+        while let Some(unit_header) = unit_iter.next()? {
+            let unit = self.dwarf.unit(unit_header)?;
+
+            let comp_dir = PathBuf::from(from_utf8(unit.comp_dir.as_ref().unwrap()).unwrap());
+
+            if let Some(ref line_program) = unit.line_program {
+                let header = line_program.header();
+
+                for file_name in header.file_names() {
+                    let combined_path = self.get_path(&comp_dir, &unit, &header, file_name);
+
+                    if combined_path.map(|p| p == path).unwrap_or(false) {
+                        let mut rows = line_program.clone().rows();
+
+                        while let Some((header, row)) = rows.next_row().unwrap() {
+                            let row_path = self.get_path(
+                                &comp_dir,
+                                &unit,
+                                &header,
+                                row.file(&header).unwrap(),
+                            );
+
+                            if row_path.map(|p| p != path).unwrap_or(true) {
+                                continue;
+                            }
+
+                            if let Some(cur_line) = row.line() {
+                                if cur_line == line {
+                                    locations.push((row.address(), row.column()));
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Look for the break point location for the best match based on the column specified.
+        match locations.len() {
+            0 => Ok(None),
+            1 => Ok(Some(locations[0].0)),
+            n => {
+                println!("Found {} possible breakpoint locations", n);
```

Comment:
This should be a `log::whatever!` :) Also I like punctuation even in log statements. That's subjective again tho.

Reply:
Fixed, thanks.